src/connector/src/source/base.rs

```diff
@@ -382,31 +381,6 @@ impl ConnectorProperties {
 
 impl ConnectorProperties {
     pub fn extract(mut with_properties: HashMap<String, String>) -> Result<Self> {
-        if Self::is_new_fs_connector_hash_map(&with_properties) {
-            let connector = with_properties
-                .remove(UPSTREAM_SOURCE_KEY)
-                .ok_or_else(|| anyhow!("Must specify 'connector' in WITH clause"))?;
-            match connector.as_str() {
-                "s3_v2" => {
-                    let assume_role = with_properties.get("s3.assume_role").cloned();
-                    return Ok(ConnectorProperties::OpendalS3(Box::new(
-                        OpendalS3Properties {
-                            s3_properties: S3Properties::try_from_hashmap(with_properties)?,
-                            assume_role,
-                        },
-                    )));
-                }
-                "gcs" => {
-                    return Ok(ConnectorProperties::Gcs(Box::new(
-                        GcsProperties::try_from_hashmap(with_properties)?,
-                    )));
-                }
```

Comment:
where's the dispatch logic going

Reply:
... Do you remember why the dispatch is needed here?

Reply:
> IIUC, previously we need the if here because s3_v2 and s3 shares the same config struct, and thus the macro cannot dispatch correctly. Since we have separated types now, the special treatment seems unnecessary and confusing.

I mentioned in the PR description

Reply:
`match_source_name_str` takes care of all cases, which I think goes to this in the lowest level

https://github.com/risingwavelabs/risingwave/blob/4695ad1239b1c160228dd7bf6f473634f57c9834/src/connector/src/macros.rs#L35-L37

Reply:
[![shipshipship](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/WjlTemxb6oA4PgZFmj08/c6b7b9b8-537f-42d9-b073-87b090e7ae72/shipshipship.png)](https://app.graphite.dev/macros/risingwavelabs)

Reply:
thanks for helping me remember the structure. No more question.